### ISSUE TITLE
Setting priors raises ValueError: need at least one array to concatenate

### ISSUE THREAD

**Issue by nolanbconaway** [bambinos/bambi#427]
Hi! I was trying to follow the docs to set custom priors on some made up data but kept hitting this numpy error in weird places. Including an example, diagnostics, and the results of my investigation here:

## Example:

```python
import bambi as bmb
import pandas as pd
import numpy as np

df = pd.DataFrame(
    dict(y=[0, 1, 0, 0, 1, 0], x_1=np.random.uniform(size=(6,)), x_2=np.random.uniform(size=(6,))),
)


priors = {
    "x_1": bmb.Prior("TruncatedNormal", sigma=1, mu=0, lower=0),
    "x_2": bmb.Prior("TruncatedNormal", sigma=1, mu=0, upper=0),
}
model = bmb.Model(
    formula="y ~ x_1 + x_2", data=df, family="bernoulli", priors=priors
)

model.build()
```

```
...
~/.pyenv/versions/3.8.10/envs/jupyter/lib/python3.8/site-packages/numpy/core/shape_base.py in hstack(tup)
    343         return _nx.concatenate(arrs, 0)
    344     else:
--> 345         return _nx.concatenate(arrs, 1)
    346 
    347 

<__array_function__ internals> in concatenate(*args, **kwargs)

ValueError: need at least one array to concatenate
```

## Diagnostics:

```python
import sys

print("sys: ", sys.version)
print("np: ", np.__version__)
print("pd: ", pd.__version__)
print("bmb: ", bmb.__version__)
```

```
sys:  3.8.10 (default, Jun 24 2021, 16:50:28) 
[Clang 12.0.5 (clang-1205.0.22.11)]
np:  1.21.2
pd:  1.3.3
bmb:  0.6.3
```

## What i tried

1. [no results in project tracker](https://github.com/bambinos/bambi/issues?q=ValueError%3A+need+at+least+one+array+to+concatenate)
2. Added/removed/changed the `automatic_priors` setting.
3. Added/removed/changed the `auto_scale` setting.
4. I put one term's prior into a tuple, and _then_ the model built, but it does not look like the prior took:

```python
priors = {
    ("x_1",): bmb.Prior("TruncatedNormal", sigma=1, mu=0, lower=0),
    "x_2": bmb.Prior("TruncatedNormal", sigma=1, mu=0, upper=0),
}
model = bmb.Model(
    formula="y ~ x_1 + x_2", data=df, family="bernoulli", priors=priors
)

model.build()
model.plot_priors()
```

![image](https://user-images.githubusercontent.com/1770896/139928119-12dd7056-37e9-4682-a96f-b89d7d5113b6.png)

You can see that x_1 is not given the correct prior whereas x_2 is.

5. Setting auto_scale=False on the _model_ allows everything to build, but the priors are not correct:

```python
model = bmb.Model(
    formula="y ~ x_1 + x_2", data=df, family="bernoulli", priors=priors, auto_scale=False
)
model.build()
model.plot_priors()
```

![image](https://user-images.githubusercontent.com/1770896/139930140-205af353-aeaf-461d-bca1-5bcda40280aa.png)

This feels like a numpy incompatibility- are these any known such issues?

**Comment by tomicapretto** [comment]
Hi @nolanbconaway 

This is definitely a bug in how we compute the automatic priors. The problem is related to how Bambi tries to compute the automatic prior for the Intercept. You could add a manual prior for the Intercept too, and it should work.

Something along these lines

```python
import bambi as bmb
import pandas as pd
import numpy as np

df = pd.DataFrame(
    dict(y=[0, 1, 0, 0, 1, 0], x_1=np.random.uniform(size=(6,)), x_2=np.random.uniform(size=(6,))),
)

# Note the Intercept here
priors = {
    "Intercept": bmb.Prior("Normal", mu=0, sigma=2),
    "x_1": bmb.Prior("TruncatedNormal", sigma=1, mu=0, lower=0),
    "x_2": bmb.Prior("TruncatedNormal", sigma=1, mu=0, upper=0),
}
model = bmb.Model(
    formula="y ~ x_1 + x_2", data=df, family="bernoulli", priors=priors
)

model.build()
```

or you can just remove the intercept if you don't want it `formula="y ~ 0 + x_1 + x_2"`.

**Comment by nolanbconaway** [comment]
That did it!! Thanks @tomicapretto ! I'm happy to set my intercept prior but also good to put this on the group's radar :)

**Comment by tomicapretto** [comment]
Glad it helped! And thanks for raising this issue. I wasn't aware of it. I'll try to fix it for the next release :)

**Comment by alexjonesphd** [comment]
Hey @tomicapretto, I'm experiencing a similar issue to @nolanbconaway here but using a hierarchical model that has some complex random effects, eg:

`y ~ a + b + c + (1|stimuli) + (1 + a + b + c|participant)`

Including a prior for the intercept doesn't help here and I get the same array concatenation error. Would a fix be manually specifying all priors? I tried adding the fixed effects of `a, b, c` in the `.set_priors()` method but noticed that the priors went from very wide to more or less a standard normal instead of what I specified.  Any advice hugely appreciated!

**Comment by tomicapretto** [comment]
Hi @alexjonesphd 

Could you share a reproducible example so I can try to see what is the problem on my side?

**Comment by alexjonesphd** [comment]
Hey @tomicapretto a closer look revealed I made a mistake - was using the splat operator to pass arguments to `bm.Prior`. However, I can confirm that using the `.set_priors()` method works really well to avoid the concatenation error!

**Comment by tomicapretto** [comment]
Great to see your problem is solved! :D 

Btw this is still a bug I need to fix. It is the next thing after #426 gets merged.

**Comment by alexjonesphd** [comment]
Sounds great! And a huge thank you to you and the rest of the devs for this package, it is truly fantastic!

**Comment by tomicapretto** [comment]
This is fixed in #430
